In [50]:
import pandas as pd
import numpy as np

from sklearn.pipeline import Pipeline
from sklearn.model_selection import KFold, cross_val_score, GridSearchCV
from sklearn.model_selection import train_test_split 
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import confusion_matrix, accuracy_score

import warnings
warnings.filterwarnings('ignore')


In [51]:
!pip install ucimlrepo

In [52]:
from ucimlrepo import fetch_ucirepo   
breast_cancer_wisconsin_original = fetch_ucirepo(id=15) 
   
X = breast_cancer_wisconsin_original.data.features 
y = breast_cancer_wisconsin_original.data.targets 
  
 


In [53]:
X.columns

Index(['Clump_thickness', 'Uniformity_of_cell_size',
       'Uniformity_of_cell_shape', 'Marginal_adhesion',
       'Single_epithelial_cell_size', 'Bare_nuclei', 'Bland_chromatin',
       'Normal_nucleoli', 'Mitoses'],
      dtype='object')

In [54]:
y['Class'].unique()

array([2, 4], dtype=int64)

In [55]:
X.isnull().sum().sum()

16

In [56]:
X.drop('Bare_nuclei',axis=1,inplace=True)

In [57]:
y['Class'].value_counts()

Class
2    458
4    241
Name: count, dtype: int64

In [58]:
y['Class'] = np.where(y['Class']==2,0,1)

In [59]:
y['Class'].value_counts()

Class
0    458
1    241
Name: count, dtype: int64

In [60]:
X_train,X_test,y_train,y_test = train_test_split(X,y,random_state=24,test_size=0.3)

In [61]:
lr = LogisticRegression()
lr.fit(X_train,y_train)
y_pred = lr.predict(X_test)
print(y_test['Class'].values)
print(y_pred)
#pd.crosstab(y_test['Class'].values, y_pred)     #confusion matrix

[0 1 1 1 1 0 1 0 0 0 1 1 1 0 0 0 0 1 1 1 0 0 1 0 0 1 0 0 0 0 1 1 0 0 0 0 0
 1 0 0 0 0 0 0 1 0 1 1 1 0 0 0 1 0 0 0 0 1 0 0 1 0 1 0 0 1 1 0 0 1 0 0 0 0
 0 0 0 1 0 0 0 0 0 1 0 0 1 0 0 0 0 1 1 1 0 0 0 1 0 0 1 0 1 0 1 1 0 1 1 1 1
 0 0 0 0 1 0 0 1 0 1 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 0 1 0 1 0 0 0 0 1
 0 1 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 1 0 0 0 0 0 0 1 0 1 0
 0 1 1 1 0 1 0 0 1 1 0 1 1 0 0 1 0 1 0 0 0 1 0 0 1]
[0 1 1 1 1 0 1 0 0 0 1 1 1 0 0 0 0 1 1 1 0 0 1 0 0 1 0 0 0 0 1 1 0 0 0 0 0
 1 0 0 0 0 0 0 1 0 1 1 1 0 0 0 1 0 0 0 0 1 0 0 1 0 1 0 0 1 1 0 0 1 0 0 0 0
 0 0 0 1 0 0 0 0 0 1 0 0 1 0 0 0 0 1 1 1 0 0 0 1 0 0 1 0 0 0 1 1 0 1 1 1 1
 0 0 0 0 1 0 0 1 0 1 0 0 1 0 0 0 0 0 0 0 0 0 1 1 0 1 0 1 0 1 0 0 0 0 0 0 1
 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 1 0 1 0
 0 1 1 1 0 1 1 0 1 1 0 1 1 0 0 1 0 1 0 1 0 0 0 0 1]


col_0    0   1
row_0         
0      135   3
1        4  68

In [66]:
#(135+68)/(135+68+3+4)

0.9666666666666667

In [64]:
print (confusion_matrix(y_test['Class'], y_pred))

[[135   3]
 [  4  68]]


In [65]:
accuracy_score(y_test['Class'], y_pred)

0.9666666666666667

In [67]:
y['Class'].value_counts(normalize=True)   #458/699,241/699,  458(predicted 0+predicted 1)

Class
0    0.655222
1    0.344778
Name: proportion, dtype: float64

In [68]:
#y['Class'].value_counts

<bound method IndexOpsMixin.value_counts of 0      0
1      0
2      0
3      0
4      0
      ..
694    0
695    0
696    1
697    1
698    1
Name: Class, Length: 699, dtype: int32>

In [72]:
y_pred= np.zeros(210)

In [73]:
y_test['Class'].value_counts()

Class
0    138
1     72
Name: count, dtype: int64

In [74]:
accuracy_score(y_test['Class'],y_pred)

0.6571428571428571

Any ML Model score should be better than score of baseline model.
Here baseline (naive) model score = 0.6571428571428571 & ML Model score = 0.9666666666666667.
Hence we can say that this ML can be a relevant model.
